beacon-chain/sync/fork_watcher.go

```diff
@@ -0,0 +1,105 @@
+package sync
+
+import (
+	"github.com/pkg/errors"
+	types "github.com/prysmaticlabs/eth2-types"
+	"github.com/prysmaticlabs/prysm/beacon-chain/core"
+	"github.com/prysmaticlabs/prysm/beacon-chain/p2p"
+	"github.com/prysmaticlabs/prysm/shared/p2putils"
+	"github.com/prysmaticlabs/prysm/shared/params"
+	"github.com/prysmaticlabs/prysm/shared/slotutil"
+)
+
+// Is a background routine that observes for new incoming forks. Depending on the epoch
+// it will be in charge of subscribing/unsubscribing the relevant topics at the fork boundaries.
+func (s *Service) forkWatcher() {
+	slotTicker := slotutil.NewSlotTicker(s.cfg.Chain.GenesisTime(), params.BeaconConfig().SecondsPerSlot)
+	for {
+		select {
+		case currSlot := <-slotTicker.C():
+			currEpoch := core.SlotToEpoch(currSlot)
+			if err := s.checkForNextEpochFork(currEpoch); err != nil {
+				log.WithError(err).Error("Unable to check for fork in the next epoch")
```

Comment:
Add a comment that if a node restarts and is already past the fork epoch, it will know how to correctly subscribe to topics and there is no risk to missing this tick